pandas/_libs/tslibs/period.pyx

```diff
@@ -2687,7 +2697,7 @@ class Period(_Period):
     Period('2012-01-01', 'D')
     """
 
-    def __new__(cls, value=None, freq=None, ordinal=None,
+    def __new__(cls, value=None, freq=None, is_period=None, ordinal=None,
```

Comment:
the other typing (stubtests) issue is caused by this

`period.pyx` `__new__` has `is_period` as an argument, but `period.pyi` doesn't

is it necessary to add this argument? wouldn't it always be True anyway? I think it should be possible to remove it, and then internally, when you call `to_offset`, you can just pass `is_period=True`